### ISSUE TITLE
Workspaces & LiveEdit

### ISSUE THREAD

**Issue by paulirish** [GoogleChrome/devtools-docs#30]
- workspaces. how do they work
- WS & Sass
- LiveEdit is a gateway drug into workspaces, but its also a complement when you're using them
  - its not clear how many people start with which feature
- include links to Remy's stuff.
- what "[VM] file (322)" means
- When changing a file, why it adds \* to the end of filename and what that means
- Explain saving: Ctrl/Cmd + S vs right click and Save/Save as...

**Comment by eslachance** [comment]
Am I correct in thinking that it's still not possible to create local copies of remote files by folder? I have to open each file and hit Save As, one after the other?

**Comment by Garbee** [comment]
Correct. Saving whole folders is not currently supported.

**Comment by chisholmd** [comment]
When I added persistance I seem to have lost the ability to set break points

**Comment by eslachance** [comment]
@Garbee Will it be eventually possible? Because that would be an awesome feature.

**Comment by Garbee** [comment]
@eslachance I don't think there are any current plans for the team to add the functionality. Best thing to do is submit a feature request on the [issue tracker](http://crbug.com/new) against DevTools and see what they say.

**Comment by TristanBrotherton** [comment]
In the documentation I see edits made in the elements panel should persist and save immediately. I'm unable to create this behaviour though, and instead have to edit the relevant file under "sources" and save it. Is there still away to persist changes made in the elements tab? (otherwise you lose the ability to see immediate changes).

**Comment by paulirish** [comment]
sounds like your workspace isn't mapped.​

**Comment by TristanBrotherton** [comment]
Thanks for the quick reply Paul, If I understand correctly my work space is mapped as I can control click on a property in the elements tab, to be taken the sources tab, where I can edit and save the changes.

**Comment by JohnONolan** [comment]
@paulirish I have the same issue as @TristanBrotherton - is there any way to debug this? As Tristan says, workspace is mapped, and edits to the sources panel persist to the file system correctly. The only missing thing is that changes in the elements panel do not sync or persist to the source panel in any way.

**Comment by paulirish** [comment]
cc @aslushnikov

**Comment by aslushnikov** [comment]
@JohnONolan @TristanBrotherton hi guys, I've an idea of what might be happening, but at first lets see if I understand your scenarios correctly:

1. You author your styles in LESS
2. You transpile LESS files into CSS files with sourcemaps
3. In DevTools, you setup a workspace, which maps your CSS stylesheet to CSS filesystem files, and LESS sources from sourcemap to LESS files on the filesystem.
4. In elements panel, you see CSS styles. If you ctrl-click on the property, you're brought to the LESS file in the sources panel.

In this case editing styles through elements panel actually edits CSS - which is instant, but this doesn't change your LESS file. Whereas if you edit a LESS file in Sources panel, the editing is not instant.

**Comment by TristanBrotherton** [comment]
@aslushnikov That sounds accurate. Sounds like its a workflow issue.  My misunderstanding from reading the docs was that it would update the LESS file.

**Comment by JohnONolan** [comment]
@aslushnikov Yes. In my case, PostCSS rather than LESS, but everything else is the same.

The main things I note are:

- Changes to the PostCSS/LESS file in the sources panel *do* write to the file system, but are *not* visible when switching to the elements panel.
- Changes to the elements panel instantly appear in the browser, but do *not* carry over to the sources panel *or* write to file.

Everything in the sources panel is working exactly right - no issues at all - it's just the elements panel which seems to be disconnected, somehow, from the sources.

As a point of reference, when the workspace is mapped to the compiled CSS file rather than the source PostCSS/LESS file, then the elements/sources panels are synced correctly and as expected.

Does this help at all?

**Comment by Suncatcher** [comment]
> Correct. Saving whole folders is not currently supported.

Does the situation changed after 2 years? Is it still not supported?

**Comment by Garbee** [comment]
@Suncatcher Nope. There isn't an issue for it to be tracked and implemented by the team when they have time. This isn't a place the team looks for work. If someone wants that functionality they'll need to file a bug at [the chromium bug tracker](https://crbug.com).

**Comment by Suncatcher** [comment]
> If someone wants that functionality they'll need to file a bug at the chromium bug tracker.

Okay, I [fixed](https://bugs.chromium.org/p/chromium/issues/detail?id=675894) this horrible deficiency)

**Comment by bulgariamitko** [comment]
+1 i want this feature

**Comment by JaganJonnala** [comment]
+1
I want this feature

**Comment by userlond** [comment]
> @Garbee
> I think I did not explain understand, so please look images
> ![111](https://cloud.githubusercontent.com/assets/3394927/4911083/6a858d3c-6488-11e4-8cd3-65e84930dd08.jpg)
> ![222](https://cloud.githubusercontent.com/assets/3394927/4911087/7189de08-6488-11e4-83c8-9e594ccfd58c.jpg)

Same problem, I think it will never be fixed.
Bug in bugtracker: https://bugs.chromium.org/p/chromium/issues/detail?id=596037
